### PR TITLE
Fix issue where uploading files would hang if a request failed while …

### PR DESCRIPTION
**Pull request by kevinli7**
…using stream_slice.StreamSlice.

### CODE REVIEW

**Comment by vilasj** [comment]
There are some lint failures.

File: apitools/base/py/transfer_test.py [hunk 1]
```diff
@@ -477,3 +484,77 @@ def testMediaCompressed(self):
...
+                self.response,  # Initial request in InitializeUpload().
+                self.fail_response,  # 503 status code from server.
+                self.response # Successful request.
+            ])
+
+            # Initialization.
+            upload.InitializeUpload(self.request, bytes_http)
+            upload.StreamInChunks()
+
+            # Ensure the mock was called three times
```
Comment (vilasj) [inline apitools/base/py/transfer_test.py]: Why three? A short comment may be helpful.

Also, would it be useful to assert anything else?

Comment (kevinli7) [reply apitools/base/py/transfer_test.py]: Do you think another comment is necessary here. In the side_effect above, I've commented on the what the responses are.

Comment (vilasj) [reply apitools/base/py/transfer_test.py]: Ah ok. May be extract responses into a local variable and use the len of that in the assert?

Comment (kevinli7) [reply apitools/base/py/transfer_test.py]: Done.

**Pull request merged**